src/Auditor.jl

```diff
@@ -1,5 +1,19 @@
 export audit, collect_files, collapse_symlinks
 
+# Logging functions.  TODO: improve them using Julia logging functionalities, or
+# third-party packages like `LoggingExtras`.
+function audit_warn(short::AbstractString, long::AbstractString, filename::AbstractString, line_number::Int; show_long::Bool = true)
```

Comment:
I kind of think we should have an `@audit_warn` that auto-collects `@__LINE__` and `@__FILE__` and can accept `show_long` as a kwarg.  So we would invoke as:

```
@audit_warn "this is my message" show_long=true
```

Reply:
I quickly hacked (to be improved)
```julia
macro audit_warn(long, short = long)
    return quote
        println("##vso[task.logissue type=warning;sourcepath=", $(String(__source__.file::Symbol)),
                ";linenumber=", $(__source__.line), ";]", $(esc(short)))
        @warn($(esc(long)))
    end
end
```
which does the job for the `println`, but reports the wrong file:line for the `@warn` call :disappointed:

Reply:
Instead of `@warn` try directly calling `Base.Logging.logmsg_code((__module__, String(__source__.file), __source__.line), :Warn, $(esc(long)))`

I got that from figuring out what `@warn` actually lowers to: https://github.com/JuliaLang/julia/blob/2d5741174ce3e6a394010d2e470e4269ca54607f/base/logging.jl#L214

Reply:
Yes, right before going to have dinner I tried something along these lines:
```julia
macro audit_warn(long, short = long)
    return quote
        println("##vso[task.logissue type=warning;sourcepath=", $(String(__source__.file::Symbol)),
                ";linenumber=", $(__source__.line), ";]", $(esc(short)))
        Base.CoreLogging.logmsg_code((__module__, String(__source__.file), __source__.line), :Warn, $(esc(long)))
    end
end
```
But I get
```
ERROR: LoadError: UndefVarError: __source__ not defined
Stacktrace:
 [1] top-level scope at /tmp/foo.jl:5
 [2] include at ./boot.jl:328 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1105
 [4] include(::Module, ::String) at ./Base.jl:31
 [5] include(::String) at ./client.jl:424
 [6] top-level scope at REPL[39]:1
```
Need to figure out hygiene here.

Reply:
Try this:
```
macro audit_warn(long, short = long)
    return quote
        println("##vso[task.logissue type=warning;sourcepath=", $(String(__source__.file::Symbol)), ";linenumber=", $(__source__.line), ";]", $(esc(short)))
        Base.CoreLogging.logmsg_code($(__module__), $(String(__source__.file)), $(__source__.line), :Warn, $(esc(long)))
    end
end
```